src/Symfony/Component/DependencyInjection/Loader/Configurator/Traits/ArgumentTrait.php

```diff
@@ -22,6 +30,25 @@
      */
     final public function args(array $arguments)
     {
+        $arguments = array_map(
+            function ($argument) {
+                if ($argument instanceof DecoratedServiceArgument) {
+                    $decorated = $this->definition->getDecoratedService();
+
+                    if (!$decorated) {
+                        throw new \LogicException(
+                            sprintf('Specify decorates() for the "%s" service', $this->id)
+                        );
+                    }
+
+                    return new ReferenceConfigurator($decorated[1] ? $decorated[1] : $this->id . '.inner');
```

Comment:
".inner" is not hardcoded when decorating services, see "renamedId".

Reply:
I don't get it. `$decorated[1]` contains value of `renamedId`. I use `.inner` suffix when user didn't specify `$renamedId`.

Reply:
oh ok!